### PR TITLE
Gra 1198 enrollment keys

### PR DESCRIPTION
**Pull request by 0xdcarns**
Do **not** merge until post 0.18.1
Use server branch https://github.com/gravitl/netmaker/tree/GRA-1198-enrollment_keys
Testing steps:

- [x] Create an enrollment key for each: uses, expiration, and unlimited
- [x] Add different networks to two of them
- [x] Ensure that the one with networks registered, also adds healthy nodes to the host that registered (daemon must be running)
- [x] Ensure that the uses one works
- [x] Ensure the expiration one works
- [x] Ensure that the unlimited one works
- [x] Ensure that: when the expiration one expires, you can not register 
- [x] Ensure that when the uses are gone on the uses one, you can not register
- [x] Ensure when the unlimited one is deleted, you can not register
- [x] Ensure that if you use both, you join both networks (nodes are created) for the different networks

### CODE REVIEW

**Review by afeiszli — APPROVED** [review]

File: cmd/register.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	Use:   "register",
+	Short: "register to a Netmaker instance",
+
+	Long: `register to a Netmaker instance using: 
+token: netclient register -t <token> // join using an enrollment token`,
+	Run: func(cmd *cobra.Command, args []string) {
+		flags := viper.New()
+		flags.BindPFlags(cmd.Flags())
+		// CLI should always take password from stdin
+		flags.Set("readPassFromStdIn", true)
```
Comment (mattkasun) [inline cmd/register.go:24]: what is point of setting this flag?

File: cmd/register.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+// registerCmd represents the register command
+var registerCmd = &cobra.Command{
+	Use:   "register",
+	Short: "register to a Netmaker instance",
+
+	Long: `register to a Netmaker instance using: 
+token: netclient register -t <token> // join using an enrollment token`,
+	Run: func(cmd *cobra.Command, args []string) {
+		flags := viper.New()
```
Comment (mattkasun) [inline cmd/register.go:21]: lines 21- 29 should be
token, err := cmd.Flags().GetString("token")
if err != nil {
    cmd.Usage()
    return
}

**Review by mattkasun — CHANGES_REQUESTED** [review]

File: cmd/register.go (hunk 1 above)
Comment (0xdcarns) [reply cmd/register.go:24]: no idea, I copied it from join

**Comment by theguy951357** [comment]
I am having trouble getting nodes successfully connected on this one.
I have netmaker and netclient built on the branches.
I use the create enrollment key api call mentioned on GRA-1198-enrollment_keys 
I then use the token created to call netclient register -t <token> on two nodes.
nodes show up on the UI.
netclient list returns "no such network"
nodes cannot ping eachother.

**Review by theguy951357 — APPROVED** [review]
checkmarks pass on the pr

critical pr list 100%

pass

**Pull request merged**